server/src/utils/callAutomationUtils.ts

```diff
@@ -0,0 +1,294 @@
+};
+
+/**
+ * This is how we attach call automation to the call. We want to not start transcription here. but
+ * will start it later when the call connects.
+ * @param serverCallId - this can be fetched from the client call object with the handler in the '
+ * Call info object.
+ */
+export const connectRoomsCall = async (serverCallId: string): Promise<void> => {
+  const transcriptionOptions = {
+    transportUrl: callAutomationConfig?.ServerWebSocketUrl ?? '',
+    transportType: 'websocket',
+    locale: 'en-US',
+    startTranscription: false
+  };
+
+  const connectionId = Object.keys(CALLCONNECTION_ID_TO_CORRELATION_ID).find((key) =>
+    CALLCONNECTION_ID_TO_CORRELATION_ID[key].serverCallId.includes(serverCallId)
+  );
+
+  /**
+   * Check if the call automation client and connection for the call already exists. If it does, we don't need to create a new one.
+   */
+  if (connectionId) {
+    console.log('Call connection for call already exists:', connectionId);
+    return;
+  }
+
+  const options = {
+    callIntelligenceOptions: {
+      cognitiveServicesEndpoint: callAutomationConfig?.CognitionAPIEndpoint ?? ''
+    },
+    transcriptionOptions: transcriptionOptions
+  };
+  const res = await getCallAutomationClient().connectCall(
+    {
+      kind: 'serverCallLocator',
+      id: serverCallId
+    },
+    callAutomationConfig?.ServerHttpUrl ?? '',
+    options
+  );
+  console.log('Connect call result', res);
+  const callConnection = res.callConnection;
+  console.log('Call connection', callConnection);
+};
+
+export const startTranscriptionForCall = async (
+  callConnectionId: string,
+  options?: TranscriptionOptions
+): Promise<void> => {
+  console.log('Starting transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
+  const callConnection = await getCallAutomationClient().getCallConnection(callConnectionId);
+
+  return await callConnection.getCallMedia().startTranscription(options);
+};
+
+export const stopTranscriptionForCall = async (callConnectionId: string): Promise<void> => {
+  console.log('Stopping transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
+  const callConnection = await getCallAutomationClient().getCallConnection(callConnectionId);
+  return await callConnection.getCallMedia().stopTranscription();
+};
+
+/**
+ * We should be storing this based on the connectionId and not the correlationId
+ * We should re-write all the fetches based on the callId to use the serverCallId. This is
+ * because all of the clients that join can have different callId's and we need to make sure we are
+ * pulling the correct transcription data.
+ */
+export const TRANSCRIPTION_STORE: { [key: string]: Partial<CallTranscription> } = {};
+/**
+ * Used to map between the call connection id and the correlation id from both transcription and
+ * call automation events.
+ */
+export const CALLCONNECTION_ID_TO_CORRELATION_ID: {
```

Comment:
There's a lot of mutable global variables in this file. Consider writing a Manager class that encapsulates a lot of these raw entries, and then could also make them private but expose accessor functions, perform any lifecycle management etc.
(Can consider as a future improvement, don't need to block this PR on it).